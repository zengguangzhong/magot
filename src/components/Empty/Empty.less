@import '../../variable.less';

.@{prefix}-empty {
  position: relative;
  text-align: center;
  line-height: @empty-line-height;
  .box();
  .box-sizing();
}

.@{prefix}-empty-image {
  height: @empty-image-height;
  margin-bottom: @empty-image-margin-bottom;
  &.only-empty {
    margin-bottom: 0;
  }
  & > img {
    height: 100%;
  }
}

.@{prefix}-empty-tips {
  font-size: @empty-tips-font-size;
  color: @empty-tips-color;
  .box();
}
