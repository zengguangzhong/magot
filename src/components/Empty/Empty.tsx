import React from 'react';
import cx from 'classnames';

import * as component from '../component';

import './Empty.less';

export interface EmptyProps
  extends component.BaseComponent,
    component.NestedComponent {
  /**
   * 设置自定义图片地址，默认是Magot的本地图片
   */
  image?: string;

  /**
   * 自定义图片样式
   */
  imageStyle?: React.CSSProperties;

  /**
   * 是否显示图片
   * @default true
   */
  imageVisible?: boolean;

  /**
   * 提示文案
   * @default 暂无数据
   */
  tips?: string | React.ReactNode;

  /**
   * 自定义提示文案样式
   */
  tipsStyle?: React.CSSProperties;
}

const defaultProps: Partial<EmptyProps> = {
  image: require('./empty.png'),
  imageVisible: true,
  tips: '暂无数据',
};

function Empty(props: EmptyProps) {
  const prefix = component.getComponentPrefix('empty');
  const cls = component.getComponentClasses('empty', props);
  return (
    <div className={cls} style={props.style}>
      {props.imageVisible && props.image && (
        <div
          className={cx(prefix + '-image', {
            'only-empty': !props.tips && !props.children,
          })}
          style={props.imageStyle}>
          <img src={props.image} alt="NO-DATA" />
        </div>
      )}
      {props.tips && (
        <p className={prefix + '-tips'} style={props.tipsStyle}>
          {props.tips}
        </p>
      )}
      {props.children}
    </div>
  );
}

Empty.defaultProps = defaultProps;

export default Empty;
