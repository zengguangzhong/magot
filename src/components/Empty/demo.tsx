import React from 'react';
import Empty from './Empty';
import Button from '../Button';
import { Link } from 'react-router-dom';

function EmptyDemo() {
  return (
    <>
      <Link to="/" className="demo-goback">
        返回
      </Link>
      <div className="demo-box">
        <Empty />
      </div>
      <div className="demo-box">
        <Empty
          image="https://oss3.rabbitpre.com/spa/img/6237031221cdb5b0eab70443c546cb79.png"
          imageStyle={{ height: 250, marginBottom: 0 }}
          tipsStyle={{
            color: '#fff',
            fontSize: 30,
            position: 'absolute',
            top: 110,
            width: '100%',
          }}
          style={{ backgroundColor: '#383838' }}
        />
      </div>
      <div className="demo-box">
        <Empty imageVisible={false} />
      </div>
      <div className="demo-box">
        <Empty tips="" />
      </div>
      <div className="demo-box">
        <Empty
          tips="您目前没有创建任何游戏"
          tipsStyle={{ color: '#999', fontSize: 24, margin: '15px 0' }}>
          <Button type="primary" icon="plus">
            新建游戏
          </Button>
        </Empty>
      </div>
    </>
  );
}

export default EmptyDemo;
