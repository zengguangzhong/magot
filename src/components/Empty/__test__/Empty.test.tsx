import React from 'react';
import { render } from 'enzyme';
import Empty from '../Empty';
import Button from '../../Button';
import { injectStyleTestSuites } from '../../../../test/shared.test';

describe('Empty', () => {
  it('should renders dom correctly', () => {
    const wrapper = render(<Empty />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders `image` and `imageStyle` props when pass in', () => {
    const image =
      'https://oss3.rabbitpre.com/spa/img/6237031221cdb5b0eab70443c546cb79.png';
    const wrapper = render(
      <Empty image={image} imageStyle={{ height: 250 }} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').attr('src')).toEqual(image);
    expect(wrapper.find('.mgt-empty-image').attr('style')).toEqual(
      'height:250px'
    );
  });

  it('should renders `tips` and `tipsStyle` props when pass in', () => {
    const tips = '您目前没有创建任何游戏';
    const wrapper = render(
      <Empty tips="您目前没有创建任何游戏" tipsStyle={{ color: '#999' }} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-empty-tips').text()).toEqual(tips);
    expect(wrapper.find('.mgt-empty-tips').attr('style')).toEqual('color:#999');
  });

  it('should renders image hidden', () => {
    const wrapper = render(<Empty imageVisible={false} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-empty-image')).toHaveLength(0);
  });

  it('should renders non-tips', () => {
    const wrapper = render(<Empty tips="" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-empty-tips')).toHaveLength(0);
  });

  it('should renders have children', () => {
    const wrapper = render(
      <Empty>
        <Button type="primary" icon="plus">
          新建游戏
        </Button>
      </Empty>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-button')).toHaveLength(1);
  });

  injectStyleTestSuites(Empty);
});
