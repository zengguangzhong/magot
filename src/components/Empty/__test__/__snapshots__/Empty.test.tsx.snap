// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty should renders \`image\` and \`imageStyle\` props when pass in 1`] = `
<div
  class="mgt-empty"
>
  <div
    class="mgt-empty-image"
    style="height:250px"
  >
    <img
      alt="NO-DATA"
      src="https://oss3.rabbitpre.com/spa/img/6237031221cdb5b0eab70443c546cb79.png"
    />
  </div>
  <p
    class="mgt-empty-tips"
  >
    暂无数据
  </p>
</div>
`;

exports[`Empty should renders \`tips\` and \`tipsStyle\` props when pass in 1`] = `
<div
  class="mgt-empty"
>
  <div
    class="mgt-empty-image"
  >
    <img
      alt="NO-DATA"
      src="test-file-stub"
    />
  </div>
  <p
    class="mgt-empty-tips"
    style="color:#999"
  >
    您目前没有创建任何游戏
  </p>
</div>
`;

exports[`Empty should renders custom className 1`] = `
<div
  class="mgt-empty jest-test-class"
>
  <div
    class="mgt-empty-image"
  >
    <img
      alt="NO-DATA"
      src="test-file-stub"
    />
  </div>
  <p
    class="mgt-empty-tips"
  >
    暂无数据
  </p>
</div>
`;

exports[`Empty should renders custom styles 1`] = `
<div
  class="mgt-empty"
  style="color:#333;background-color:#f0f0f0"
>
  <div
    class="mgt-empty-image"
  >
    <img
      alt="NO-DATA"
      src="test-file-stub"
    />
  </div>
  <p
    class="mgt-empty-tips"
  >
    暂无数据
  </p>
</div>
`;

exports[`Empty should renders dom correctly 1`] = `
<div
  class="mgt-empty"
>
  <div
    class="mgt-empty-image"
  >
    <img
      alt="NO-DATA"
      src="test-file-stub"
    />
  </div>
  <p
    class="mgt-empty-tips"
  >
    暂无数据
  </p>
</div>
`;

exports[`Empty should renders have children 1`] = `
<div
  class="mgt-empty"
>
  <div
    class="mgt-empty-image"
  >
    <img
      alt="NO-DATA"
      src="test-file-stub"
    />
  </div>
  <p
    class="mgt-empty-tips"
  >
    暂无数据
  </p>
  <button
    class="mgt-button mgt-button-primary"
    type="button"
  >
    <i
      class="mgt-icon mgt-icon-plus left"
    />
    新建游戏
  </button>
</div>
`;

exports[`Empty should renders image hidden 1`] = `
<div
  class="mgt-empty"
>
  <p
    class="mgt-empty-tips"
  >
    暂无数据
  </p>
</div>
`;

exports[`Empty should renders non-tips 1`] = `
<div
  class="mgt-empty"
>
  <div
    class="mgt-empty-image only-empty"
  >
    <img
      alt="NO-DATA"
      src="test-file-stub"
    />
  </div>
</div>
`;
